### ISSUE TITLE
Image attachments in announcements do not show up

### ISSUE THREAD

**Issue by fonglh** [Coursemology/coursemology2#1557]
Images uploaded when creating announcements are base64 encoded and stored directly in the code.

However, they do not show up when viewed. The `<img>` tag is empty.

The data could be getting stripped out on display.

**Comment by weiqingtoh** [comment]
Although it is [declared in the code that Announcements can allow for image uploads](https://github.com/Coursemology/coursemology2/blob/master/app/models/course/announcement.rb#L4), the feature is still not fully supported yet. 

This declaration can be found in more details here -> ([has_one/many_attachment(s) extension](https://github.com/Coursemology/coursemology2/blob/master/lib/extensions/attachable/active_record/base.rb))
Currently the feature works well with file uploads, which result in a creation of a URL following the regex found in the above extension:
```ruby
ATTACHMENT_ID_REGEX = /\/attachments\/([0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12})$/
```

Summernote provides image uploads ([docs](http://summernote.org/deep-dive/#onimageupload)) but the default handler is `base64 dataURL on IMG tag`. Hence, it is necessary to modify the javascript to support our uploading to S3 (as well as our `attachment` extension). It is suggested to read through the following issue (https://github.com/summernote/summernote/issues/72) for more details.

**Comment by fonglh** [comment]
I'll prefer just using Summernote's built in mechanism for images instead of using S3 attachments for image uploads.

That simplifies our code and reduces the code paths which puts stuff on S3.

There is some elegance in having everything be an attachment, but I'm not sure it's worth the work writing custom Javascript to allow Summernote to do it.

**Comment by fonglh** [comment]
Another workaround is to upload the image to the workbin then link to its URL from Summernote. No code changes required.

**Comment by weiqingtoh** [comment]
After some reading, `base64` encoding is only recommended to be done on smaller images. Although the image can now be served with the html text (thereby reducing the number of http requests), there is some tradeoff in terms of the size (typically there is a 10% gain). This is definitely not favourable in the long run when we have image-heavy announcements. 

Links:
 - http://www.askapache.com/online-tools/base64-image-converter/ (`base64 encoding adds around 10% to the image size after gzip compression`)
 - http://davidbcalhoun.com/2011/when-to-base64-encode-images-and-when-not-to/ (Some experiments)
 - https://varvy.com/pagespeed/base64-images.html (It is best to only base64 encode tiny images. If you try to base64 larger images you will end up with a great deal of code in your html and lose out on the performance benefits.)

We should hook in the javascript at some point: 
 - I don't think the amount of additional code required is significant (versus benefits) ([See here](https://github.com/summernote/summernote/issues/72#issuecomment-27676684))
 - We can extend this functionality to most HTML text columns
 - We shouldn't encode images in our database (especially if there are large ones, ie. 10mb) 

For now we can offer the workbin image upload as a short-term workaround.

**Comment by fonglh** [comment]
The workbin image upload doesn't work in the announcement email notification. Not sure about the S3 or base64 methods.

**Comment by pottiepie** [comment]
> Another workaround is to upload the image to the workbin then link to its URL from Summernote. No code changes required.

This will clutter the workbin if we have lots of images to put in announcements. And it looks quite silly having an "Images" workbin

**Comment by allenwq** [comment]
Fixed after #1857
